packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js

```diff
@@ -36,14 +119,50 @@ export default function SidebarNavigationScreenNavigationMenus() {
 		},
 		[ history ]
 	);
+
+	if ( hasResolvedNavigationMenus && ! hasNavigationMenus ) {
+		return (
+			<SidebarNavigationScreenWrapper>
+				{ __( 'There are no Navigation Menus.' ) }
+			</SidebarNavigationScreenWrapper>
+		);
+	}
+
+	if ( ! hasResolvedNavigationMenus || isLoading ) {
+		return (
+			<SidebarNavigationScreenWrapper>
+				<div className="edit-site-sidebar-navigation-screen-navigation-menus__placeholder" />
+				<div className="edit-site-sidebar-navigation-screen-navigation-menus__placeholder" />
+				<div className="edit-site-sidebar-navigation-screen-navigation-menus__placeholder" />
+			</SidebarNavigationScreenWrapper>
+		);
+	}
+
 	return (
-		<SidebarNavigationScreen
-			title={ __( 'Navigation' ) }
-			content={
-				<div className="edit-site-sidebar-navigation-screen-navigation-menus">
-					<NavigationInspector onSelect={ onSelect } />
+		<BlockEditorProvider
+			value={ publishedInnerBlocks }
+			onChange={ onChange }
+			onInput={ onInput }
+		>
+			<SidebarNavigationScreenWrapper
+				actions={
+					<Inserter
+						position="bottom right"
+						isAppender
+						selectBlockOnInsert={ false }
+						shouldDirectInsert={ false }
+						__experimentalIsQuick
+						toggleProps={ { as: SidebarButton } }
+					/>
+				}
+			>
+				<div className="edit-site-sidebar-navigation-screen-navigation-menus__content">
+					<NavigationMenuContent
+						innerBlocks={ publishedInnerBlocks }
```

Comment:
I think now that we're actually rendering a hidden `BlockList` we might not need to wait for "innerBlocks" loading and there's also some action calls that could be removed from `NavigationMenuContent`. I kept them for now but I'll see later if I can get rid of them and simplify further.